assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { RichText } from '@wordpress/block-editor';
+import { Button, Spinner } from '@wordpress/components';
+import { RawHTML, useEffect, useRef, useState } from '@wordpress/element';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { PostSelector } from '../../components';
+import useElementClickDetector from './element-click-detector';
+
+const AttachmentContent = ( props ) => {
+	const [ selectedPost, setSelectedPost ] = useState( null );
+	const [ failedToFetch, setFailedToFetch ] = useState( false );
+	const [ searchValue, setSearchValue ] = useState( '' );
+	const [ isFetching, setIsFetching ] = useState( false );
+	let fetchRequest, isStillMounted;
+
+	const {
+		attributes,
+		setAttributes,
+		toggleAttachment,
+	} = props;
+
+	const {
+		attachmentClass,
+		postId,
+		postType,
+		title,
+		wrapperStyle,
+	} = attributes;
+
+	const fetchSelectedPost = () => {
+		isStillMounted = true;
+		if ( postId ) {
+			setIsFetching( true );
+			const currentFetchRequest = fetchRequest = apiFetch( {
+				path: `/wp/v2/${ postType }s/${ postId }`,
```

Comment:
Not all post types work this way. The path of the rest api doesn't have to match the post type here.

Reply:
FWIW the post selector is currently hardcoded to only support `page` and `post`, for which this assumption here is true.

Is your recommendation to make this future-proof by fetching the respective REST base from `wp-json/wp/v2/types`?

Reply:
I would just pass the values of post types / path to rest api in a `wp_localize_script`. I would also add a filter to allow users to add custom post types.

Reply:
Makes sense. We have the system in place for that already 👍